File: testing_codes/C++/18-AllocateWithMalloc.cpp
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int size;
    int* arr;

    printf("Enter the size of the array: ");
    scanf("%d", &size);

    arr = (int*)malloc(size * sizeof(int))

    if (arr == NULL)
    {
        printf("Memory allocation failed.\n");
        return 1;s
    }

    printf("Memory allocated successfully.\n");


    free(arr); 

    return 0;
}

File: testing_codes/Java/18-AllocateWithMalloc.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the array: ");
        int size = scanner.nextInt();
        int[] arr = new int[size];

        System.out.println("Memory allocated successfully.");

        // Do something with the allocated memory here...

        // Freeing memory is handled automatically by Java's garbage collector.
    }
}
